packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -760,6 +772,18 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
 
     private initBlobsCache(blobs: IBlob[]) {
         blobs.forEach((blob) => this.blobCache.set(blob.id, blob));
+        this.scheduleClearBlobsCache();
+    }
+
+    /**
+     * Stop the current timer for clearing the blob cache (if any) and schedule a new one
+     */
+    private scheduleClearBlobsCache() {
+        const blobCacheTimeoutDuration = 10000;
```

Comment:
This looks fine for starters, but something to keep in mind -continuously setting & resetting timer might be expensive.
In similar places in Office we developed a pattern of building small virtualization layer on top of "hardware" timer where we would not move "hardware" timer if it is too far out, thus amortizing all set/reset calls into O(1).

(I can tell a story what kind of bugs we hit whiteout it, finding some interesting scalability bugs in Windows).

Reply:
I'm suspecting we would bottleneck in the blob-reading code before it bottlenecks on the timer set/clear.  But if it comes up we can deal with it then.